scripts/setup-lxd.sh

```diff
@@ -1,12 +1,27 @@
-#!/bin/sh
+#!/bin/bash
 # Copyright 2016 Canonical Ltd.
 # Licensed under the AGPLv3, see LICENCE file for details.
-set -ex
+set -eux
 
 # Do the manual steps a user has to run on a fresh system to get an lxd
 # bridge so the juju lxd provider can function. Taken from changes made
 # to cloud-init to do approximately this.
 
+VERSION=$(lxd --version)
+
+# LXD 2.3+ needs lxdbr0 setup via lxc.
+if [[ "$VERSION" > "2.2" ]]; then
+    if [[ ! $(lxc network list | grep lxdbr0) ]]; then
+        # Configure a known address ranges for lxdbr0.
+        lxc network create lxdbr0 \
+            ipv4.address=10.0.8.1/24 ipv4.nat=true \
```

Comment:
We don't know that this network is free unless you check it, right? Same for the IPv6 one, for which I am sure there is at least one RFC about how to create a subnet on a host.

I would avoid IPv6 + NAT if at all possible since the containers should be able to perform some form of IPv6 autoconfiguration and communicate with the rest of the network. Bridges don't need addresses either, so we should be able to drop that. Of course if 'lxc network create' needs that configuration we will have to provide it, but some form of algorithmically generated (probably link local) address would be better.

On the subject of addresses, I don't recognise 0xd42 as an IPv6 address prefix. Did it come from keyboard bashing? See https://en.wikipedia.org/wiki/IPv6_address#Unicast_addresses